hammer_time/hammer_time.py

```diff
@@ -33,12 +34,37 @@ def perform(client):
         remove_and_wait(client, new_status.iter_new_machines(old_status))
 
 
+def choose_machine(client):
+    """Choose a machine from the client's model at random.
+
+    :param client: The ModelClient to get machines for.
+    :return: a machine-id.
+    :raises: InvalidActionError if there are no machines to choose from.
+    """
+    status = client.get_status()
+    machines = list(m for m, d in status.iter_machines(containers=False))
+    if len(machines) == 0:
+        raise InvalidActionError('No machines to choose from.')
+    return choice(machines)
+
+
+class RebootMachineAction:
+    """Action that reboots a machine."""
+
+    def generate_parameters(client):
+        return {'machine_id': choose_machine(client)}
+
+    def perform(client, machine_id):
+        """Add and remove many containers using the cli."""
+        old_status = client.get_status()
+        client.juju('ssh', (machine_id, 'sudo', 'reboot'), check=False)
```

Comment:
This opp has proven to be awkward to test. This call requests a reboot, but I don't see a confirmation that a reboot happened. Some tests poll uptime and ports to be certain the machine did reboot and did comeback.